File: Interview Preparation Kit - CPP/01. Warm-up Challenges/004. Repeated String.cpp
// Problem: https://www.hackerrank.com/challenges/repeated-string/problem
// Score: 20


#include <iostream>
#include <string>

using namespace std;


long long count(string const & str){
    long long ans = 0;
    for (const char & c: str){
        ans += c == 'a';
    }
    return ans;
}


int main(){
    string str;
    cin >> str;

    long long n;
    cin >> n;

    long long ans = 0;
    ans = n / str.size() * count(str) + count(str.substr(0, n % str.size()));

    cout << ans;
    return 0;
}

File: Interview Preparation Kit - Python/01. Warm-up Challenges/004. Repeated String.py
# Problem: https://www.hackerrank.com/challenges/repeated-string/problem
# Score: 20


def repeated_string(s, n):
    return n // len(s) * s.count('a') + s[0: n % len(s)].count('a')


s = input()
n = int(input())
print(repeated_string(s, n))
